cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -209,6 +221,14 @@ var validAwsRefIdRegex = regexp.MustCompile(fmt.Sprintf(`^aws\:\/\/\/[-0-9a-z]*\
 // AwsRefFromProviderId creates AwsInstanceRef object from provider id which
 // must be in format: aws:///zone/name
 func AwsRefFromProviderId(id string) (*AwsInstanceRef, error) {
+	// Special case for SageMaker format: aws:///<region>/sagemaker/...
+	if strings.HasPrefix(id, "aws:///") && strings.Contains(id, "/sagemaker") {
```

Comment:
Can we use a pattern match here instead of prefix check?

Reply:
Done